session_srtcp.go

```diff
@@ -112,28 +112,50 @@ func (s *SessionSRTCP) decrypt(buf []byte) error {
 		return err
 	}
 
+	forward := func(pkt rtcp.Packet) (closed bool, err error) {
+		for _, ssrc := range pkt.DestinationSSRC() {
+			r, isNew := s.session.getOrCreateReadStream(ssrc, s, newReadStreamSRTCP)
+			if r == nil {
+				return true, nil // Session has been closed
+			} else if isNew {
+				s.session.newStream <- r // Notify AcceptStream
+			}
+
+			readStream, ok := r.(*ReadStreamSRTCP)
+			if !ok {
+				return false, fmt.Errorf("failed to get/create ReadStreamSRTP")
+			}
+
+			_, err = readStream.write(decrypted)
+			if err != nil {
+				return false, err
+			}
+		}
+		return false, nil
+	}
+
 	pkt, err := rtcp.Unmarshal(decrypted)
 	if err != nil {
 		return err
 	}
 
-	for _, ssrc := range pkt.DestinationSSRC() {
-		r, isNew := s.session.getOrCreateReadStream(ssrc, s, newReadStreamSRTCP)
-		if r == nil {
-			return nil // Session has been closed
-		} else if isNew {
-			s.session.newStream <- r // Notify AcceptStream
-		}
-
-		readStream, ok := r.(*ReadStreamSRTCP)
-		if !ok {
-			return fmt.Errorf("failed to get/create ReadStreamSRTP")
-		}
-
-		_, err = readStream.write(decrypted)
+	cpkt := rtcp.CompoundPacket(pkt)
+	if cpkt.Validate() == nil {
+		_, err = forward(&cpkt)
 		if err != nil {
 			return err
 		}
+	} else {
+		for _, p := range pkt {
```

Comment:
Shouldn't an rtcp-rsize packet with multiple sub-packets should be delivered as one `[]Packet` rather than split up? Sending two packets in one datagram might have some semantic meaning in the application.

Reply:
hmmm... may have to take another look at the spec. do all of the packets in an rtcp-rsize packet share a set of DestinationSSRC?

Reply:
Good question. I believe the whole compound packet goes to the same SSRC but I'm not sure on the rsize ones.

Reply:
i've just force-pushed a revision that sends the whole mess to a superset of `Packet.DestinationSSRC()` for each packet in an rtcp-rsize packet (that isn't also a valid `CompoundPacket`)

Reply:
Also, test coverage for srtp is pretty bad, but i don't really have time to look at it today. I'm going to file an issue.